src/Symfony/Component/Cache/Adapter/ChainAdapter.php

```diff
@@ -73,7 +73,9 @@ static function ($sourceItem, $item, $sourceMetadata = null) use ($defaultLifeti
                 $item->isHit = $sourceItem->isHit;
                 $item->metadata = $item->newMetadata = $sourceItem->metadata = $sourceMetadata;
 
-                if (0 < $defaultLifetime) {
+                if (isset($item->metadata[CacheItem::METADATA_EXPIRY])) {
+                    $item->expiresAt(\DateTime::createFromFormat('U.u', $item->metadata[CacheItem::METADATA_EXPIRY]));
```

Comment:
Just update to latest release with this change and it broke my site. As i know the expiry value is always a float number without precision (microsecond part). Uses `U.u` causing problem here. The format should be `U`. Can you advice?

Reply:
Can you elaborate why `U.u` casing a problem?

It seams to work: https://3v4l.org/pdadj

Reply:
please open an issue with the error message you see and a reproducer or a failing test case ideally

Reply:
I suppose we should use `sprintf('%.3F', ...)`
the issue happens when the expiry is an int

Reply:
Yes, in my case expiry was int, not float.
```
simple.cache:
        default_lifetime: 86400
        adapters:
          - cache.adapter.apcu
          - cache.adapter.redis
```

```
public function isPartner(User $user): bool
    {
        return $this->simpleCache->get(md5((string)$user->getId() . __METHOD__), function (ItemInterface $item) use ($user) {
            $item->expiresAfter(86400 * 7); // 1 week
            $item->tag(md5((string)$user->getId()));
            
           // compute
            return false;
        });
    }
```

Reply:
thanks, would you mind sending a PR with the fix and a test case, please?
for brach 4.4

Reply:
It is middle-night my time and I would do it tomorrow if it is convenient.

Reply:
It is hard to me to reproduce myself today and finally i found down the problem could happen with ArrayAdapter in a race condition making expiry to be int. I'm not sure how could it happen like that. BTW, the PR is at #38879.